src/main/java/org/gradle/profiler/studio/launcher/StudioConfigurationProvider.java

```diff
@@ -0,0 +1,108 @@
+package org.gradle.profiler.studio.launcher;
+
+import com.dd.plist.NSDictionary;
+import com.dd.plist.NSObject;
+import com.dd.plist.NSString;
+import com.dd.plist.PropertyListParser;
+import com.google.common.collect.ImmutableMap;
+import org.gradle.profiler.OperatingSystem;
+
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class StudioConfigurationProvider {
+
+    public static StudioConfiguration getLaunchConfiguration(Path studioInstallDir) {
+        if (OperatingSystem.isMacOS()) {
+            return getMacOSConfiguration(studioInstallDir);
+        } else {
+            return getWindowsOrLinuxConfiguration(studioInstallDir);
+        }
+    }
+
+    private static StudioConfiguration getWindowsOrLinuxConfiguration(Path studioInstallDir) {
+        String mainClass = "com.intellij.idea.Main";
+        List<Path> classPath = Stream.of("lib/bootstrap.jar", "lib/util.jar", "lib/jdom.jar", "lib/log4j.jar", "lib/jna.jar")
```

Comment:
Can we parse the classpath on Windows/Linux from a file as well? It feels wrong to hardcode this here.

Reply:
We discussed this and it is too complicated to parse it from the start script. Let's see if the need arises to change it and then let's re-visit.